### ISSUE TITLE
Dimension mob health scaling

### ISSUE THREAD

**Issue by Cruuk** [Tslat/Advent-Of-Ascension#3048]
Yes, of course I know all the dimension stuff is gonna look completely different when it gets re-added in 1.18+, this is more of a "what I don't want to see in revamped dims" thing.

This is a tangent based on the 2-3 playthroughs I've done with post-3.2 AoA by now, and I think I can say that the health scaling for mobs is a bit too high and "homogenous" if that makes any sense. This topic has a lot to it, so I'll try my best to elaborate.

The current mob scaling doesn't exactly make things harder, it just makes mobs tedious health sponges. Even if the mobs were high quality, it wouldn't instantly fix this issue. I remember watching a streamer play AoA, and they said that games should either have lots of low HP enemies, or small amounts of large HP enemies, doing both high HP and high spawnrates at the same time isn't the best idea usually.
Compared to weapons and mob damage, mob HP scales very sharply, and I know you want lategame dims to be hard, but it's a bit much.

However, it's not as simple as that, I think "homogenous" is a better way to describe it. The issue is mainly exacerbated with small/flying ranged mobs. The baseline for mob HP is too tight. Something like Fungat has 80 HP, only 10 less HP than Fungback, even though Fungat is smaller and more annoying to fight.
This issue is at its worst with flying ranged mobs, especially smaller stuff like Polytom and Omnilight, whose HP is barely different from the easier-to-hit mobs. It is absolutely obnoxious wasting several ammo hitting/missing flying ranged mobs, only for 2 more to spawn after you spent 2 minutes killing 1. I guess there is a reason Ghast only has 10 HP. It incentivizes me to just run through dims until I find a mob that's less annoying to kill compared to the small HP sponge flying mobs.

I could be biased with the mobs being low quality, but would high quality mobs fix this issue if they were still HP sponges that spawned in groups and were also small in size.
Even with melee mobs, it's very easy for a swarm to group up with the high spawn rates, and having to slowly cut through a group of 100 HP mobs while kiting is annoying- especially when the dozen small 95 HP flying melee mobs join in, with the weapons following a more linear progression system against sharp HP scaling.

It forces me to move in specific ways with trying to aggro as little mobs as possible, and if you intend for that to be an aspect of lategame challenge, I wouldn't mind it too much, but this is prevalent through the whole mod.

Even as soon as t1, the HP scaling starts sharply from t0, this is mainly prevalent in Nether where AoA Nether mobs are annoying to take down, not because they're challenging, but because they take several hits, even when you get Emberstone stuff. Stuff like Snipers also become flat out worse lategame, as everything big and small increases in HP to such a high degree that Snipers can barely do much other than weaken 1 mob in a group of 5 before you fight them.

I do believe it's important to give the player feedback that they're progressing by having the "weaker" stuff take less effort to kill compared to the "stronger" stuff. This isn't black and white, of course it would be too easy if mob HP scaled evenly with weapons, but it's currently too excessive. I don't have an issue with larger mobs having high HP, but when medium/small mob HP is barely any different from large mobs and also spawn very frequently, it just slows things down. Would it make accessing world tables "too easy" with being able to kill mobs faster? No not at all, that's a silly thing to think about. Lowering item drop chances to "make up" for faster-to-kill mobs would also benefit nobody, let people have some fun.

Notice how the "regular sized" mob only has 10 less HP than the big huge mob.
![image](https://user-images.githubusercontent.com/38159897/160811423-e4af5178-3da1-491e-89b7-438ec82b94cd.png)

Or how the small, dangerous, frequent flying mobs have HP no different than a beefy melee mob
![image](https://user-images.githubusercontent.com/38159897/160811907-109e5f1e-4548-417b-9179-13d55385d86c.png)

------------------------------------------

To summarize, mob HP post-OW starts a bit too high and in general scales a bit too much sharply and doesn't add much for good difficulty. This is mainly problematic with small/flying mobs, whose HP is too homogenous with larger mobs. Everything sharing high HP is also made even worse by spawnrates still being frequent.

Spawnrates would be fine if everything wasn't a tank. Introduce a more dramatic HP difference between small and big mobs. Stuff like ranged/flying mobs also don't need high HP with the mobility they already have. Large mobs can keep the high HP , as it makes sense and they're not the main source behind this problem
